### PR TITLE
177585974/Category pages

### PR DESCRIPTION
**Pull request by teallarson**
TO DO:
* Add to "solutions" nav menu (will complete after [this pr](https://github.com/grouparoo/www.grouparoo.com/pull/428) is approved with updated nav.

DONE:
* Refactor existing [id] template in solutions to allow both category and comparison pages in the `/solutions` directory
* Add content for Reverse ETL page
* Add content for CDP page
* Add in-page nav
* Replace current placeholder dividers with relevant content/graphics
* Fine tune/re-import visuals created in Figma
* Header image/banner

### CODE REVIEW

**Comment by tealjulia** [comment]
TO DO: 
* Add content for CDP page
* Weigh and pick: header image/feature banner/in-page nav
* Replace current placeholder dividers with relevant content/graphics
* Update (and improve speed on) testing of solutions pages

File: components/solutions/featureBanner.tsx [hunk 1]
```diff
@@ -24,8 +26,7 @@ export default function SectionDivider({
           <Row style={{ textAlign: "center" }}>
             <Col md={12}>
               <h2>
-                Grouparoo is the customer data platform that meets you where you
-                are at.
+                Grouparoo is the {category} that meets you where you are at.
```
Comment (andyjih) [inline components/solutions/featureBanner.tsx]:
```suggestion
                Grouparoo is the {category} that meets you where you are.
```

File: pages/blog/using-sync-modes.mdx [hunk 2]
```diff
@@ -0,0 +1,118 @@
+---
```
Comment (andyjih) [inline pages/blog/using-sync-modes.mdx:1]: Somehow this blogpost got pulled into your PR. Did some branches get mixed up at some point?

File: pages/solutions/reverse-etl.mdx [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+At Grouparoo, we’re helping to lead that trend, so we thought we’d share an overview of what Reverse ETL is, and how Grouparoo enables businesses to use their data to the fullest extent.
+
+### What is Reverse ETL?
+
+Before we discuss Reverse ETL, let’s talk about “regular” ETL.
+
+ETL is a data warehouse term which stands for Extract, Transform, and Load. ETL describes the process of taking data from a database or source (Extract), modifying it or cleaning it up (Transform), and then putting it in a data warehouse (Load).
+
+<SectionDivider
+  imagePath="/images/home/solutions/categories/etl.svg"
```
Comment (andyjih) [inline pages/solutions/reverse-etl.mdx:45]: Not a blocker, but one of the more important aspects of ETL is that it's pulling data from a product database as well as data from SaaS tools like Salesforce. I'd suggest we replace one of the logos with a database logo or icon, but looks good overall!

File: pages/solutions/customer-data-platform.mdx [hunk 4]
```diff
@@ -0,0 +1,104 @@
+---
+category: "Customer Data Platform"
+pros:
+  [
+    "give companies a single, unified view of their data",
+    "allow dynamic grouping",
+    "syndicate groups to other SAAS tools",
```
Comment (andyjih) [inline pages/solutions/customer-data-platform.mdx]:
```suggestion
    "syndicate groups to other SaaS tools",
```

**Review by andyjih — CHANGES_REQUESTED** [review]
Overall looks good! A few minor tweaks. 

Separately, it looks like Pedro's blog post about sync modes somehow got pulled into your branch...?

File: pages/solutions/reverse-etl.mdx (hunk 3 above)
Comment (teallarson) [reply pages/solutions/reverse-etl.mdx:45]: Sounds good.  I'll pop that into the tracker.

**Review by andyjih — APPROVED** [review]
👍  Nice work!

File: pages/blog/using-sync-modes.mdx (hunk 2 above)
Comment (teallarson) [reply pages/blog/using-sync-modes.mdx:1]: Likely.  I am trying to think how to resolve that without messing up main?

File: components/footer.tsx [hunk 5]
```diff
@@ -128,6 +128,12 @@ export default function Footer() {
                 <Link href="/solutions/hightouch-alternative">
                   <a>Hightouch Alternative</a>
                 </Link>
+                <Link href="/solutions/reverse-etl">
+                  <a>Reverse ETL</a>
+                </Link>
+                <Link href="/solutions/customer-data-platform">
+                  <a>Customer Data Platform</a>
+                </Link>
```
Comment (evantahler) [inline components/footer.tsx]: `<br/>`s are missing here

File: pages/solutions/reverse-etl.mdx [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+    },
+    {
+      title: "📈  Improve advertising impact",
+      text: "With a Reverse ETL, you can automate list creation for effective advertising campaigns -- saving you advertising dollars and time.",
+    },
+    {
+      title: "🛍  Market more meaningfully",
+      text: "Help your marketing team create campaigns based on real, meaningful data.",
+    },
+  ]
```
Comment (evantahler) [inline pages/solutions/reverse-etl.mdx:29]: Nice!

File: utils/inflectors.ts [hunk 7]
```diff
@@ -9,6 +9,13 @@ export function titleize(str:string) {
     });
 }
 
+// Take a title string and replace all spaces with commas, convert to lowercase
+
+export function urlize(str:string): string {
+
+  return str.replace(/\ /g, '-').toLowerCase();
+}
+
```
Comment (evantahler) [inline utils/inflectors.ts:18]: This is actually a thing called `slugify`, as you are making URL 'slugs'!  Not that we need them, but there are whole packages dedicated to handling this kind of thing in multiple languages, etc.

**Review by evantahler — CHANGES_REQUESTED** [review]
Looking good, but there are still a few things to fix:

1. There are some line-breaks missing in the footer

Screen Shot 2021-05-07 at 7 05 32 PM

2. Is it intended for the gradient call outs to stretch all the way to the page border?  We do this on the homepage.  currently they end at the container width. 

Screen Shot 2021-05-07 at 7 06 50 PM

3. There's a lot of other things in this PR (blog post, comparison pages, new fancy navigation, etc). We may be able to rebase away some of the older commits that don't belong.  This guide may help https://medium.davidendersby.me/til-git-rebasing-removing-rage-commits-4b5c42e7af3a

**Comment by teallarson** [comment]
Closed in favor of PR #432

**Pull request closed without merging**